tests/Doctrine/Tests/DBAL/Functional/BlobTest.php

```diff
@@ -55,6 +57,43 @@ public function testInsert()
         self::assertEquals(1, $ret);
     }
 
+    public function testInsertProcessesStream()
+    {
+        $this->_conn->insert('blob_table', [
+            'id'          => 1,
+            'clobfield'   => 'ignored',
```

Comment:
Please test this as well.

Reply:
So you want to process streams also for `ParameterType::STRING`?

Of course we could do, but then I don't really get the purpose of the parameter types... On `SELECT`, only the `LARGE_OBJECT` type will return a stream, whereas `STRING` is a... string.

Reply:
No, you should be able to bind a bind a character stream to a CLOB field. `LARGE_OBJECT` doesn't imply *B*LOB, it implies _any_ LOB.

Reply:
> On `SELECT`, only the `LARGE_OBJECT` type will return a stream, whereas `STRING` is a... string.

I think this is an issue we should fix in 3.0. Regardless of whether it's TEXT or BLOB, it should be returned as a stream.

Reply:
I think the test covers it now?